### DISCUSSION TITLE
Does gnark plan to add secret-sharing related primitive

### DISCUSSION THREAD

**Discussion by zhiqiangxu** [Consensys/gnark#1394] [Q&A]
Secret sharing plays a crucial role in MPC (Multi-Party Computation). `gnark` is primarily positioned as a high-performance zk-SNARK library, but is there a plan to incorporate secret-sharing-related primitives in the future?

**Comment by ivokub** [comment]
We have [Groth16](https://github.com/Consensys/gnark/pull/1372) and [KZG](https://github.com/Consensys/gnark-crypto/pull/589) key generation protocols as open PRs. We currently to not plan to add anything else related to MPC.

**Reply by ivokub** [reply]
And those PRs we try to merge soon, it takes a bit time to review though.
